zerver/lib/events.py

```diff
@@ -63,17 +64,22 @@ def user_data(row: Dict[str, Any]) -> Dict[str, Any]:
         )
 
         is_admin = row['is_realm_admin']
-
-        return dict(
-            email=row['email'],
-            user_id=row['id'],
-            avatar_url=avatar_url,
-            is_admin=is_admin,
-            is_bot=row['is_bot'],
-            full_name=row['full_name'],
-            timezone=row['timezone'],
-            is_active = row['is_active'],
-        )
+        person = dict(email=row['email'],
+                      user_id=row['id'],
+                      avatar_url=avatar_url,
+                      is_admin=is_admin,
+                      is_bot=row['is_bot'],
+                      full_name=row['full_name'],
+                      timezone=row['timezone'],
+                      is_active = row['is_active'])
+        for field in custom_fields:
+            try:
+                value = CustomProfileFieldValue.objects.get(
+                    user_profile=get_user_profile_by_id(row['id']), field=field).value
+                person[field.name] = value
+            except CustomProfileFieldValue.DoesNotExist:
+                person[field.name] = None
+        return person
```

Comment:
This will pass the values of custom fields to `person` object, so that in future we can get custom field's value of any user not only current user in frontend.

Reply:
Yeah, so I think attaching to `person` is correct.  I think we probably don't want to attach as `person[field.name]`, though, because then someone bad could name a field "user_id" or "email" and cause all manner of chaos.

So I think we might want something more like `person['profile'][field.name]` as the shape.

Also, this does database queries in a loop, which is a huge perf problem (hence the giant jumps in query counts below).

We want this to do a bulk query for all the users in question...